sql/core/src/main/scala/org/apache/spark/sql/jdbc/TeradataDialect.scala

```diff
@@ -31,4 +31,19 @@ private case object TeradataDialect extends JdbcDialect {
     case BooleanType => Option(JdbcType("CHAR(1)", java.sql.Types.CHAR))
     case _ => None
   }
+
+  override def isCascadingTruncateTable(): Option[Boolean] = Some(false)
+
+  /**
+   * The SQL query used to truncate a table.
```

Comment:
Please add more description about Teradata's lack of `TRUNCATE` syntax here. Otherwise, people try to make a PR for this mistakenly.